### PR TITLE
Adding a firewall update location

### PR DESCRIPTION
**Pull request by michael-carter-instaclustr**

### CODE REVIEW

File: instaclustr/api_client.go [hunk 1]
```diff
@@ -168,6 +168,21 @@ func (c *APIClient) CreateFirewallRule(data []byte, clusterID string) error {
 	return nil
 }
 
+func (c *APIClient) UpdateFirewallRule(data []byte, clusterID string) error {
+	url := fmt.Sprintf("%s/provisioning/v1/%s/firewallRules/", c.apiServerHostname, clusterID)
+	resp, err := c.MakeRequest(url, "PUT", data)
+	if err != nil {
+		return err
+	}
+	bodyText, err := ioutil.ReadAll(resp.Body)
```
Comment (Mitchel-Stig-Instaclustr) [inline instaclustr/api_client.go:177]: slight optimisation - could we move this inside the below if statement so we dont do this on success if its not needed

File: instaclustr/resource_firewall_rule.go [hunk 2]
```diff
@@ -78,15 +72,26 @@ func resourceFirewallRuleCreate(d *schema.ResourceData, meta interface{}) error
...
+
+	jsonStr, err := makeFirewallRuleJson(d)
 	if err != nil {
 		return fmt.Errorf("[Error] Error creating firewall rule: %s", err)
 	}
 
 	err = client.CreateFirewallRule(jsonStr, d.Get("cluster_id").(string))
 	if err != nil {
-		return fmt.Errorf("[Error] Error creating firewall fule: %s", err)
+		return fmt.Errorf("[Error] Error creating firewall rule: %s", err)
```
Comment (Mitchel-Stig-Instaclustr) [inline instaclustr/resource_firewall_rule.go:94]: Ha!

**Review by Mitchel-Stig-Instaclustr — CHANGES_REQUESTED** [review]

File: instaclustr/api_client.go (hunk 1 above)
Comment (michael-carter-instaclustr) [reply instaclustr/api_client.go:177]: Not as it is, 'cos the body Text is put in the  error message

Comment (Jack-Kilrain) [reply instaclustr/api_client.go:177]: I think Mitch means to put the `ioutil.ReadAll(...)` in the if check below. Also, I think we should also handle checking the `err` var incase the body is unreadible:
```go
if resp.StatusCode != 202 {
	bodyText, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		bodyText = "<unparsable body>"
	}
	return errors.New(fmt.Sprintf("Status code: %d, message: %s", resp.StatusCode, bodyText))
}
```

**Review by arjunrajinstaclustr — APPROVED** [review]

**Review by Mitchel-Stig-Instaclustr — APPROVED** [review]

**Review by arjunrajinstaclustr — APPROVED** [review]

**Review by Jack-Kilrain — APPROVED** [review]

**Pull request merged**